src/types.ts

```diff
@@ -1,65 +1,189 @@
+/**
+ * macOS applications can be distributed via the Mac App Store (MAS) or directly
+ * downloaded from the developer's website. @electron/osx-sign distinguishes between
+ * MAS apps (`mas`) or non-MAS apps (`darwin`).
+ */
 export type ElectronMacPlatform = 'darwin' | 'mas';
-type SigningDistributionType = 'development' | 'distribution';
 
+/**
+ * MAS apps can be signed using Development or Distribution certificates.
+
+ * See [Apple Documentation](https://developer.apple.com/help/account/create-certificates/certificates-overview/) for more info.
+ */
+export type SigningDistributionType = 'development' | 'distribution';
+
+/**
+ * @interface
+ * @internal
```

Comment:
`@internal` is a typedoc-specific directive used to hide this type from the generated documentation interface.

This is applied to most types in this file since end users don't need to know the implementation details of how we generate `SignOptions` and `FlatOptions`.